cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/UsageHandlerTest.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.services.http.handlers;
+
+import co.cask.cdap.internal.app.services.http.AppFabricTestBase;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import org.apache.http.HttpResponse;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.InputStreamReader;
+import java.util.Set;
+
+/**
+ * Tests for {@link co.cask.cdap.gateway.handlers.UsageHandler}
+ */
+public class UsageHandlerTest extends AppFabricTestBase {
+
+  private static final Gson GSON = new Gson();
+
+  @Test
+  public void testProgramStreamUsage() throws Exception {
+    Id.Application app = Id.Application.from("somespace", "noapp");
+    Id.Program program = Id.Program.from(app, ProgramType.FLOW, "noprogram");
+    Id.Stream stream = Id.Stream.from("somespace", "nostream");
+
+    Assert.assertEquals(0, getAppStreamUsage(app).size());
+    Assert.assertEquals(0, getProgramStreamUsage(program).size());
+    Assert.assertEquals(0, getStreamProgramUsage(stream).size());
+
+    getUsageDataset().register(program, stream);
```

Comment:
Yes, this test is only for the `UsageHandler`, not testing usage registry integration with apps/streams/etc.